### PR TITLE
feat: implement precision alignment and end-to-end DLRM training with Criteo dataset

### PR DESCRIPTION
**Pull request by Choimoe**
Added single-day Criteo data loader with fallback validation logic. Implemented custom preprocessing scripts and full DLRM training/test workflows:

```bash
nohup torchrun --nnodes 1 --nproc_per_node 1 --rdzv_backend c10d --rdzv_endpoint localhost --rdzv_id 54321 --role trainer tests/dlrm_main_single_day.py --single_day_mode --in_memory_binary_criteo_path ./processed_day_0_data --batch_size 1024 --learning_rate 0.005 --epochs 1 --pin_memory --mmap_mode --embedding_dim 128 --adagrad > training_output.log 2>&1
```

The last few lines of the log file:

```bash
# training_output.recstore.08310505.log
[DEBUG][op_torch] emb_read_torch: values shape=[410, 128], dtype=float, data_ptr=0x63aa80d3c400
[DEBUG][op_torch] emb_read_torch: values start with: [0, 0, 33.6794, 0, 43.2784, 0, 0, 0, 0, 0, ] [5.01694, 0, 56.9622, 0, 55.9405, 0, 0, 0, 0, 0, ] [0, 272.716, 236.158, 0, 773.966, 0, 0, 0, 0, 0, ] [0, 97.3052, 86.6881, 0, 271.506, 0, 0, 0, 0, 0, ] [0, 7295.42, 6122.17, 0, 21188.5, 0, 0, 0, 0, 0, ] [0, 355.151, 302.317, 0, 1020.74, 0, 0, 0, 0, 0, ] [0, 613.84, 525.271, 0, 1758.55, 0, 0, 0, 0, 0, ] [0, 0, 7.00742, 0, 5.00353, 0, 0, 0, 0, 0, ] [0, 40.149, 31.6143, 0, 103.57, 0, 0, 0, 0, 0, ] [51.7378, 0, 22.76, 0, 48.5221, 0, 0, 0, 0, 0, ] 
[DEBUG][op_torch] emb_read_torch: keys shape=[410], dtype=long int, data_ptr=0x79044a2fb380
[DEBUG][op_torch] emb_read_torch: embedding_dim=128
[DEBUG][op_torch] emb_read_torch: keys start with: 0, 0, 0, 0, 0, 0, 0, 18, 95, 0, 
[DEBUG][op_torch] emb_read_torch: values shape=[410, 128], dtype=float, data_ptr=0x63aa80d3c400
[DEBUG][op_torch] emb_read_torch: values start with: [0, 126.708, 129.066, 0, 323.982, 0, 0, 0, 0, 0, ] [0, 98.4864, 93.66, 0, 259.123, 0, 0, 0, 0, 0, ] [0, 1374.2, 1169.25, 0, 3950.89, 0, 0, 0, 0, 0, ] [0, 148.512, 124.701, 0, 428.077, 0, 0, 0, 0, 0, ] [0, 1042.12, 877.501, 0, 3011.72, 0, 0, 0, 0, 0, ] [6.5708, 0, 0, 0, 2.20708, 0, 0, 0, 0, 0, ] [0, 659.467, 559.841, 0, 1899.35, 0, 0, 0, 0, 0, ] [0, 32.4444, 59.9151, 0, 103.145, 0, 0, 0, 0, 0, ] [0, 4703.02, 3949, 0, 13656.1, 0, 0, 0, 0, 0, ] [0, 5488.27, 4606.98, 0, 15937.9, 0, 0, 0, 0, 0, ] 
[DEBUG][op_torch] emb_read_torch: keys shape=[410], dtype=long int, data_ptr=0x79044a2fc050
[DEBUG][op_torch] emb_read_torch: embedding_dim=128
[DEBUG][op_torch] emb_read_torch: keys start with: 29, 2, 29, 19, 18, 29, 29, 18, 19, 29, 
[DEBUG][op_torch] emb_read_torch: values shape=[410, 128], dtype=float, data_ptr=0x63aa80d3c400
[DEBUG][op_torch] emb_read_torch: values start with: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ] [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ] [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ] [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ] [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ] [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ] [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ] [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ] [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ] [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ] 

Validation: 100%|██████████| 7/7 [00:02<00:00,  2.72it/s]
Validation: 100%|██████████| 7/7 [00:02<00:00,  2.51it/s]
Epoch 1 - Validation Loss: 0.3672, Validation AUROC: 0.6459
Training completed!
```

Refactored RecStoreEBC test suite with single-table accuracy checks and improved assertions. [ebc_cpu_test.08310534.log](https://github.com/user-attachments/files/22061313/ebc_cpu_test.08310534.log)

Origin CTest: [ctest.08310539.log](https://github.com/user-attachments/files/22061316/ctest.08310539.log)

### CODE REVIEW

File: model_zoo/torchrec_dlrm/download_criteo.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    return 1
+}
+
+failed=()
+for day in $(seq 0 23); do
+    download_file "$day" || failed+=("day_${day}.gz")
+done
+
+if [ ${#failed[@]} -ne 0 ]; then
+    printf 'Failed: %s\n' "${failed[@]}"
```
Comment (Choimoe) [inline model_zoo/torchrec_dlrm/download_criteo.sh:45]: 截止到当前时间（2025年8月31日05:45:17 UTC/GMT +8.00），huggingface上的文件仍然是全部损坏的，所以目前这个脚本下载下来的文件会全部失败。可以在 [讨论区](https://huggingface.co/datasets/criteo/CriteoClickLogs/discussions) 看到有对于数据损坏的多个讨论，但是仍然没有解决方案。所以我对损坏的数据进行了截断处理。

File: model_zoo/torchrec_dlrm/tests/README.MD [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
+> based on Commit 9c31e49a4e692ea47caa6452f43124fc3cf6e678
 
 | Test Case | Test Objective | Status | Description |
 |-----------|---------------|--------|-------------|
 | test_basic_initialization | Basic initialization functionality | ✅ | Verify single table configuration initialization |
 | test_multi_table_initialization | Multi-table initialization | ✅ | Verify multi-table configuration initialization |
-| test_empty_config_error | Empty configuration error handling | ✅ | Verify exception thrown for empty configuration |
-| test_missing_config_fields | Missing configuration fields error handling | ✅ | Verify exception thrown for missing required fields |
+<!-- | test_empty_config_error | Empty configuration error handling | ✅ | Verify exception thrown for empty configuration | -->
+<!-- | test_missing_config_fields | Missing configuration fields error handling | ✅ | Verify exception thrown for missing required fields | -->
```
Comment (Choimoe) [inline model_zoo/torchrec_dlrm/tests/README.MD:13]: 这两个测试实际上没有用。对于剩下的测试去除了之前的一些特判（修掉了一些ebc的bug）。

File: model_zoo/torchrec_dlrm/tests/test_ebc_precision.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+    # b) Gradient Calculation (Implicitly tested by weight update)
+    dummy_loss_standard = standard_output_kt.values().sum()
+    dummy_loss_recstore = recstore_output_kt.values().sum()
+    
+    standard_optimizer.zero_grad()
+    sparse_optimizer.zero_grad()
+
+    dummy_loss_standard.backward()
+    dummy_loss_recstore.backward()
```
Comment (Choimoe) [inline model_zoo/torchrec_dlrm/tests/test_ebc_precision.py:143]: 这里精度对齐只是执行了一遍forward和一遍backward，后续可以进行更多测试。

File: src/framework/pytorch/op_torch.cc [hunk 4]
```diff
@@ -118,73 +118,8 @@ torch::Tensor emb_read_torch(const torch::Tensor& keys, int64_t embedding_dim) {
...
-  }
-
-  base::RecTensor rec_keys  = ToRecTensor(cpu_keys, base::DataType::UINT64);
-  base::RecTensor rec_grads = ToRecTensor(cpu_grads, base::DataType::FLOAT32);
-
-  RECSTORE_LOG(3, "[DEBUG] emb_update_torch: calling op->EmbUpdate");
-  op->EmbUpdate(rec_keys, rec_grads);
-  RECSTORE_LOG(3, "[DEBUG] emb_update_torch: EmbUpdate done");
+  throw std::runtime_error(
+      "emb_update_torch is deprecated. Use the Python-based sparse optimizer.");
```
Comment (Choimoe) [inline src/framework/pytorch/op_torch.cc:122]: 将emb_update暂时使用上层的优化器代替，便于对齐精度和修复计算模块的bug。
